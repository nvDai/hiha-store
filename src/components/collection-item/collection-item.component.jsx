import React from 'react'
import PropsTypes from 'prop-types'
import { createStructuredSelector } from 'reselect'
import { connect } from 'react-redux'

import CustomButton from '../custom-button/custom-button.component'
import LazyImage from '../lazy-image/lazy-image.component'

import { addItem } from 'redux/cart/cart.actions'
import { selectCartHidden } from 'redux/cart/cart.selectors'

import './collection-item.styles.scss'

export const CollectionItem = ({ item, addItem }) => {
	const { name, price, imageUrl } = item
	return (
		<div className='collection-item'>
			<LazyImage
				placeHolder={`${imageUrl}?tr=bl-30,q-50`}
				src={imageUrl}
				alt={name}
				className='image'
			/>
			<div className='collection-footer'>
				<span className='name'>{name}</span>
				<span className='price'>${price}</span>
			</div>
			<CustomButton
				onClick={() => {
					addItem(item)
				}}
			>
				Add to cart
			</CustomButton>
		</div>
	)
}

const mapStateToProps = createStructuredSelector({
	hidden: selectCartHidden,
})

const mapDispatchToProps = dispatch => ({
	addItem: item => dispatch(addItem(item)),
})

CollectionItem.propsTypes = {
	item: PropsTypes.object.isRequired,
	addItem: PropsTypes.func,
}

export default connect(mapStateToProps, mapDispatchToProps)(CollectionItem)
