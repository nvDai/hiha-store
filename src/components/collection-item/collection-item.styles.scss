.collection-item {
  width: 22%;
  display: flex;
  flex-direction: column;
  height: 350px;
  align-items: center;
  position: relative;

  .image {
    width: 100%;
    height: 100%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
    transition: all 0.2s ease-in;
  }

  .custom-button {
    width: 80%;
    opacity: 0.7;
    position: absolute;
    top: 50%;
    transition: all 0.2s ease-in;
    display: none;
    padding: 0;
  }

  &:hover {
    .image {
      opacity: 0.8;
    }

    .custom-button {
      opacity: 0.85;
      display: flex;
    }
  }

  .collection-footer {
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-between;
    font-size: 18px;

    .name {
      width: 90%;
      margin-bottom: 15px;
    }

    .price {
      width: 10%;
    }
  }

   @media screen and (max-width: 767px) {
  
      width: calc((100vw - 36px)/2);
      margin-bottom: 16px;
      &:nth-last-child(-n+2) {
        margin-bottom: 0;
      }

      .custom-button {
        position: relative;
        display: block;
        top: 0;
        margin-top: 12px;
        width: 100%;
        z-index: 2;
        opacity: 1;
        outline: unset;

        &:hover {
          opacity: 1;
          background-color: black;
          color: white;
          border: none;
      	}
      }

      .collection-footer .price {
        width: auto;
      }

      &:hover {
        .image {
          opacity: 1;
        }
      }
    }
}