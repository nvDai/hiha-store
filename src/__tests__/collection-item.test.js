import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { CollectionItem } from '../components/collection-item/collection-item.component'

let wrapper, props
beforeEach(() => {
	props = {
		addItem: jest.fn(),
		item: { name: 'Shirt', price: 0, imageUrl: 'shirt.jpg' },
	}
	wrapper = shallow(<CollectionItem {...props} />)
})
describe('CollectionItem', () => {
	it('should render without crashing', () => {
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})

test('CustomButton should be clicked', () => {
	wrapper.find('CustomButton').simulate('click')
	expect(props.addItem).toHaveBeenCalledTimes(1)
})
