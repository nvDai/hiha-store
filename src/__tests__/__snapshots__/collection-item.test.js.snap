// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem should render without crashing 1`] = `
<div
  className="collection-item"
>
  <LazyImage
    alt="Shirt"
    className="image"
    placeHolder="shirt.jpg?tr=bl-30,q-50"
    src="shirt.jpg"
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      Shirt
    </span>
    <span
      className="price"
    >
      $
      0
    </span>
  </div>
  <CustomButton
    onClick={[Function]}
  >
    Add to cart
  </CustomButton>
</div>
`;
